lib/ui/contact_list/chat_list_screen.dart

```diff
@@ -463,7 +475,30 @@ class _ChatListScreenState extends ConsumerState<ChatListScreen> with TickerProv
                       },
                       itemCount:
                           isInLoadingState ? _loadingSkeletonCount : filteredChatItems.length,
-                      separatorBuilder: (context, index) => Gap(8.w),
+                      separatorBuilder: (context, index) {
+                        if (isInLoadingState) {
+                          return Gap(8.w);
+                        }
+
+                        // Check if this separator should be between pinned and unpinned items
+                        final currentItem = filteredChatItems[index];
```

Comment:
It the sorting and separation was done in a provider and returned in two different list, there would be no need to filter the sorted list in the widget.  I think that way here would be simpler: the widget would just render the separator between the two lists (pinned and unpinned) without having to check for every item if the next item is pinned or unpinned.